python/ray/serve/deployment_state.py

```diff
@@ -1330,12 +1331,11 @@ def _check_curr_status(self) -> bool:
             )
             == 0
         ):
-            # Check for deleting.
-            if target_replica_count == 0 and all_running_replica_cnt == 0:
+            if self._deleting:
```

Comment:
Are we checking if `all_running_replica_cnt == 0`? t looks like we're only checking that `target_replica_count == 0` (since that's where we set `self._deleting = True`), meaning that the deployment might not be completely deleted yet.

Reply:
we are not going to delete the deployment when all_running_replica_cnt == 0 only. We will delete the deployment only when delete_deployment is called.

Reply:
I see, that makes sense since a deployment can still be alive and have 0 replicas. My concern is that this function's return value indicates whether a deployment is finished deleting, not that it's in the process of deleting. Should we check `if self._deleting and all_running_replica_cnt == 0` here? Otherwise, a deployment could have `self._deleting=True`, but still have replicas running when we return True.

Reply:
good point, added it